api/notd/block_processor.py

```diff
@@ -242,9 +257,21 @@ async def process_transaction(self, transaction: TxData, retrievedEvents: List[R
         isSwap = len(nonInterstitialFromAddresses.intersection(nonInterstitialToAddresses)) > 0
         for retrievedTokenTransfer in retrievedTokenTransfers:
             retrievedTokenTransfer.isSwap = isSwap
+        # Calculate transaction value for either weth or eth
+        transactionValue = 0
+        if transaction['value'] > 0:
+            transactionValue = transaction['value']
+        elif len(wethValues) > 0:
+            for address, wethValue in wethValues:
+                if address in nonInterstitialToAddresses:
+                    transactionValue += wethValue
         # Only calculate value for remaining
-        if transaction['value'] > 0 and not isMultiAddress:
-            valuedTransfers = [retrievedTokenTransfer for retrievedTokenTransfer in retrievedTokenTransfers if not retrievedTokenTransfer.isInterstitial and not retrievedTokenTransfer.isSwap and not isOutbound]
+        if transactionValue and not isMultiAddress:
+            # If contractAddress is a opensea market place isOutbound is still a valued transfer
+            if contractAddress in MARKETPLACE_ADDRESSES:
+                valuedTransfers = [retrievedTokenTransfer for retrievedTokenTransfer in retrievedTokenTransfers if not retrievedTokenTransfer.isInterstitial and not retrievedTokenTransfer.isSwap]
+            else:
+                valuedTransfers = [retrievedTokenTransfer for retrievedTokenTransfer in retrievedTokenTransfers if not retrievedTokenTransfer.isInterstitial and not retrievedTokenTransfer.isSwap and not retrievedTokenTransfer.isOutbound]
```

Comment:
this can be simplified by something like `(contractAddress in MARKETPLACE_ADDRESSES or not retrievedTokenTransfer.isOutbound)`